src/streaming/net/HTTPLoader.js

```diff
@@ -195,30 +188,57 @@ function HTTPLoader(cfg) {
             }
         };
 
-        const _oncomplete = function() {
-            // Update request timing info
-            requestObject.startDate = requestStartTime;
-            requestObject.endDate = new Date();
-            requestObject.firstByteDate = requestObject.firstByteDate || requestStartTime;
-            httpResponse.resourceTiming.responseEnd = requestObject.endDate.getTime();
-
-            // If enabled the ResourceTimingApi we add the corresponding information to the request object.
-            // These values are more accurate and can be used by the ThroughputController later
-            _addResourceTimingValues(httpRequest, httpResponse);
-        }
+        /**
+         * Fired when a request has been aborted, for example because the program called XMLHttpRequest.abort().
+         */
+        const _onabort = function () {
+            _onRequestEnd(true)
+        };
 
         /**
-         * Fired when an XMLHttpRequest transaction completes.
-         * This includes status codes such as 404. We handle errors in the _onError function.
+         * Fired when progress is terminated due to preset time expiring.
+         * @param event
          */
-        const _onload = function () {
-            _oncomplete();
+        const _ontimeout = function (event) {
+            let timeoutMessage;
```

Comment:
Setting `fragmentRequestTimeout`to a low value like 100 for testing, we end up in an infinite loop of requests to the same fragment. It also looks like we call `_onRequestEnd` twice for timeouts. Once in `_onloadend` and once in `_ontimeout`

Reply:
Agree, fixed